diff -ur rdma-core-35.0.old/infiniband-diags/ibsysstat.c rdma-core-35.0/infiniband-diags/ibsysstat.c
--- rdma-core-35.0.old/infiniband-diags/ibsysstat.c	2021-05-10 15:58:48.000000000 +0800
+++ rdma-core-35.0/infiniband-diags/ibsysstat.c	2021-09-24 10:59:28.653893593 +0800
@@ -41,7 +41,7 @@
 
 #include "ibdiag_common.h"
 
-#define MAX_CPUS 8
+#define MAX_CPUS 256
 
 static struct ibmad_port *srcport;
 
